File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.cpp
#include <string>
#include <cmath>

class Solution {
public:
    int numSteps(std::string s) {
        int k = 0;
        while (s != "1") {
            if (s.back() == '0') {
                s.pop_back();  
            } else {
                int i = s.length() - 1;
                while (i >= 0 && s[i] == '1') {
                    s[i] = '0'; 
                    i--;
                }
                if (i >= 0) {
                    s[i] = '1'; 
                } else {
                    s = '1' + s;
                }
            }
            k++;
        }
        return k;
    }
};

File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.py
class Solution:
    def numSteps(self, s: str) -> int:
        number = int(s,2)
        count = 0
        while number != 1:
            count += 1
            if number % 2 == 1:
                number += 1
            else:
                number //= 2
        return count
